### ISSUE TITLE
pip install poppy fails

### ISSUE THREAD

**Issue by mperrin** [spacetelescope/poppy#3]
<a href="https://github.com/jiffyclub"><img src="https://avatars0.githubusercontent.com/u/920492?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jiffyclub](https://github.com/jiffyclub)**
_Tuesday Apr 30, 2013 at 16:22 GMT_
_Originally opened as https://github.com/mperrin/poppy/issues/3_

----

I'm using pip 1.3.1 with the command `pip install poppy`. See log below:

```
------------------------------------------------------------
/Users/mrdavis/.virtualenvs/pysynphot/bin/pip run on Tue Apr 30 12:14:28 2013
Downloading/unpacking poppy

  Getting page http://pypi.python.org/simple/poppy
  URLs to search for versions for poppy:
  * http://pypi.python.org/simple/poppy/
  Getting page http://www.stsci.edu/%7emperrin/software/webbpsf
  Skipping page http://www.stsci.edu/%7emperrin/software/webbpsf/poppy-0.2.8.tar.gz (from http://pypi.python.org/simple/poppy/) because of Content-Type: application/x-tar
  Analyzing links from page http://pypi.python.org/simple/poppy/
    Found link http://pypi.python.org/packages/source/p/poppy/poppy-0.2.8.tar.gz#md5=3546bff46b1a32f3b3fc1cfa56c7c95d (from http://pypi.python.org/simple/poppy/), version: 0.2.8
    Skipping link http://www.stsci.edu/%7emperrin/software/webbpsf (from http://pypi.python.org/simple/poppy/); not a file
    Found link http://www.stsci.edu/%7emperrin/software/webbpsf/poppy-0.2.8.tar.gz (from http://pypi.python.org/simple/poppy/), version: 0.2.8
    Skipping link http://www.stsci.edu/jwst/software/webbpsf/ (from http://pypi.python.org/simple/poppy/); not a file
  Analyzing links from page http://www-int.stsci.edu/~mperrin/software/webbpsf/
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/_static/default.css (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .css
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/_static/pygments.css (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .css
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/ (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); not a file
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/intro.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/genindex.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/py-modindex.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/np-modindex.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/#documentation-for-webbpsf (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); not a file
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/_images/fig_instrument_comparison.png (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .png
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html#module-poppy (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/webbpsf.html#module-webbpsf (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/gui.html#module-webbpsfgui (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/Improved_PSFs_for_Webb.pdf (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .pdf
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/#contents (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); not a file
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/intro.html#why-a-new-jwst-psf-simulator (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/intro.html#algorithm-overview (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/intro.html#quick-start (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#software-requirements (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#installing-webbpsf-via-pypi (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#installing-webbpsf-manually (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#installing-webbpsf-development-versions-and-or-contributing-to-its-development (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#installing-the-required-data-files (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#installing-or-updating-pysynphot (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/installation.html#note-for-stsci-internal-users (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#known-issues (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#plans-for-future-releases (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#version-0-2-8 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#version-0-2-7 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#version-0-2-6 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#version-0-2-5 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#versions-0-2-1-0-2-3 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#version-0-2 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/relnotes.html#version-0-1 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/webbpsf.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/webbpsf.html#usage-and-examples (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/webbpsf.html#the-jwinstrument-generic-class (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/webbpsf.html#notes-on-specific-instruments (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/webbpsf.html#utility-functions-for-display-and-plotting (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/gui.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/gui.html#gui-controls (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html#list-of-classes (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html#wavefront (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html#optical-system (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html#optical-elements (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/poppy.html#instrument (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/available_opds.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html#ote (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html#nircam (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html#nirspec (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html#niriss (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html#miri (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/references.html#instrument-filter-throughputs (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/more_examples.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/more_examples.html#typical-usage-cases (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/more_examples.html#advanced-usage-tricks (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/sampling.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/sampling.html#key-concepts (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/sampling.html#some-useful-guidance (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/sampling.html#per-instrument-sampling-requirements (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html#planning-in-fftw3 (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html#a-comparison-of-different-fft-methods (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html#a-test-comparing-all-four-planning-methods (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html#a-comparison-of-estimate-and-measure-for-different-sizes (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html#caching-of-plans-means-that-irunning-the-same-script-a-second-time-is-much-faster (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/fft_optimization.html#the-payoff-speed-improvements-in-poppy (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/#indices-and-tables (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); not a file
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/search.html (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .html
    Skipping link http://www-int.stsci.edu/%7emperrin/software/webbpsf/_sources/index.txt (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); unknown archive format: .txt
    Skipping link http://sphinx.pocoo.org/ (from http://www-int.stsci.edu/~mperrin/software/webbpsf/); not a file
  Using version 0.2.8 (newest of versions: 0.2.8, 0.2.8)
  Downloading from URL http://pypi.python.org/packages/source/p/poppy/poppy-0.2.8.tar.gz#md5=3546bff46b1a32f3b3fc1cfa56c7c95d (from http://pypi.python.org/simple/poppy/)
  Running setup.py egg_info for package poppy

    Traceback (most recent call last):

      File "<string>", line 16, in <module>

    IOError: [Errno 2] No such file or directory: '/Users/mrdavis/.virtualenvs/pysynphot/build/poppy/setup.py'

    Complete output from command python setup.py egg_info:

    Traceback (most recent call last):

  File "<string>", line 16, in <module>

IOError: [Errno 2] No such file or directory: '/Users/mrdavis/.virtualenvs/pysynphot/build/poppy/setup.py'

----------------------------------------

Command python setup.py egg_info failed with error code 1 in /Users/mrdavis/.virtualenvs/pysynphot/build/poppy

Exception information:
Traceback (most recent call last):
  File "/Users/mrdavis/.virtualenvs/pysynphot/lib/python2.7/site-packages/pip-1.2.1-py2.7.egg/pip/basecommand.py", line 107, in main
    status = self.run(options, args)
  File "/Users/mrdavis/.virtualenvs/pysynphot/lib/python2.7/site-packages/pip-1.2.1-py2.7.egg/pip/commands/install.py", line 256, in run
    requirement_set.prepare_files(finder, force_root_egg_info=self.bundle, bundle=self.bundle)
  File "/Users/mrdavis/.virtualenvs/pysynphot/lib/python2.7/site-packages/pip-1.2.1-py2.7.egg/pip/req.py", line 1042, in prepare_files
    req_to_install.run_egg_info()
  File "/Users/mrdavis/.virtualenvs/pysynphot/lib/python2.7/site-packages/pip-1.2.1-py2.7.egg/pip/req.py", line 236, in run_egg_info
    command_desc='python setup.py egg_info')
  File "/Users/mrdavis/.virtualenvs/pysynphot/lib/python2.7/site-packages/pip-1.2.1-py2.7.egg/pip/util.py", line 612, in call_subprocess
    % (command_desc, proc.returncode, cwd))
InstallationError: Command python setup.py egg_info failed with error code 1 in /Users/mrdavis/.virtualenvs/pysynphot/build/poppy
```

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Tuesday Apr 30, 2013 at 19:46 GMT_

----

Huh, it looks like it's skipping downloading the tar.gz file for some reason. That's odd. I will readily admit to not being an expert on packaging but I thought I had this set up properly. Will investigate later this afternoon hopefully.

**Comment by mperrin** [comment]
<a href="https://github.com/douglase"><img src="https://avatars1.githubusercontent.com/u/1025951?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [douglase](https://github.com/douglase)**
_Sunday Aug 11, 2013 at 17:10 GMT_

----

I also received a setup.py missing file error while installing poppy with: 
    pip install poppy --upgrade

the problem was the naming of setup-poppy.py, renaming this file setup.py  (in the  /var/.../pip-build/poppy/ folder pip created) fixed it.

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday May 14, 2014 at 20:46 GMT_

----

Fixed in version 0.3 now available on PyPI.
